Sources/Prometheus/Histogram.swift

```diff
@@ -0,0 +1,181 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+public protocol Bucketable: AdditiveArithmetic, Comparable, Sendable {
+    var bucketRepresentation: String { get }
+}
+
+public typealias TimeHistogram = Histogram<Duration>
```

Comment:
Hmmm.... not really "Time", or is it from prom specs...? If not, then mirror the contained value maybe?

Reply:
Renamed to `DurationHistogram`